ownCloud/Settings/LogFilesViewController.swift

```diff
@@ -0,0 +1,243 @@
+		self.navigationController?.setToolbarHidden(true, animated: false)
+		OCLogger.shared.resume()
+	}
+
+	deinit {
+		Theme.shared.unregister(client: self)
+	}
+
+	// MARK: - Theme support
+
+	func applyThemeCollection(theme: Theme, collection: ThemeCollection, event: ThemeEvent) {
+		self.tableView.applyThemeCollection(collection)
+
+		if event == .update {
+			self.tableView.reloadData()
+		}
+	}
+
+	// MARK: - Table view data source / delegate
+
+	override func numberOfSections(in tableView: UITableView) -> Int {
+		return 1
+	}
+
+	override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+		return logFileEntries.count
+	}
+
+	override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+		let cell = tableView.dequeueReusableCell(withIdentifier: LogFileTableViewCell.identifier, for: indexPath) as? LogFileTableViewCell
+		let logEntry = self.logFileEntries[indexPath.row]
+		cell?.textLabel?.text = logEntry.truncatedName()
+		if let date = logEntry.creationDate, let size = logEntry.size {
+			cell?.detailTextLabel?.text = "\(self.dateFormatter.string(from: date)), \(self.byteCounterFormatter.string(fromByteCount: size))"
+		}
+
+		cell?.shareAction = { [weak self] (cell) in
+			if let indexPath = self?.tableView.indexPath(for: cell) {
+				if let name = self?.logFileEntries[indexPath.row].name {
+					self?.shareLog(with: name)
+				}
+			}
+		}
+
+		return cell!
+	}
+
+	override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+		tableView.deselectRow(at: indexPath, animated: true)
+	}
+
+	override func tableView(_ tableView: UITableView, editActionsForRowAt indexPath: IndexPath) -> [UITableViewRowAction]? {
+
+		return [
+			UITableViewRowAction(style: .destructive, title: "Delete".localized, handler: { [weak self] (_, indexPath) in
+				let logEntry = self?.logFileEntries[indexPath.row]
+				if let name = logEntry?.name {
+					self?.removeLog(with: name, indexPath: indexPath)
+				}
+			})
+		]
+	}
+
+	// MARK: - Private Helpers
+
+	private func populateLogFileList() {
+		guard let logFileWriter = OCLogger.shared.writer(withIdentifier: .writerFile) as? OCLogFileWriter else { return }
+		guard let appGroupLogsURL = OCAppIdentity.shared.appGroupLogsContainerURL else { return }
+		guard let logFiles = logFileWriter.logFiles() else { return }
+
+		for fileName in logFiles {
+			if let name = fileName as? String {
+				let logURL = appGroupLogsURL.appendingPathComponent(name)
+
+				var fileSize:Int64?
+				var creationDate:Date?
+
+				if let attributes = try? FileManager.default.attributesOfItem(atPath: logURL.path) {
+					creationDate = attributes[FileAttributeKey.creationDate] as? Date
+					fileSize = (attributes[FileAttributeKey.size] as? NSNumber)?.int64Value
+				}
+
+				let logEntry = LogEntry(name: name, size: fileSize, creationDate: creationDate)
+				self.logFileEntries.append(logEntry)
+			}
+		}
+
```

Comment:
It will probably work fine the way it is, but in case APFS will not return the log files in chronological order in the future, it would be awesome to sort them by date.

Also, I feel there's an opportunity to avoid duplicating code by removing the `LogEntry` class and instead providing an equivalent from the SDK, so that `logFiles` provides an array of f.ex. `OCLogFileRecord` instances - with properties for file size, creation date and URL.

I also see the date of each log file - and especially the way to retrieve it - as an internal implementation detail. But right now there are separate implementations to get it in both the app and the SDK.

If `OCLogFileWriter.logFiles` returned an array of `OCLogFileRecord` (or any other name, really) instances with said properties, the UI would be strictly just a consumer of data provided by the SDK.

Reply:
@felix-schwarz done